generatebundlefile/ecr.go

```diff
@@ -296,6 +296,32 @@ func (c *ecrPublicClient) shaExistsInRepository(repository, sha string) (bool, e
 	return false, nil
 }
 
+// shaExistsInRepository checks if a given OCI artifact exists in a destination repo using the sha sum.
+func (c *ecrPublicClient) tagExistsInRepository(repository, tag string) (bool, error) {
+	if repository == "" || tag == "" {
+		return false, fmt.Errorf("Emtpy repository, or tag passed to the function")
+	}
+	var imagelookup []ecrpublictypes.ImageIdentifier
+	imagelookup = append(imagelookup, ecrpublictypes.ImageIdentifier{ImageTag: &tag})
+	ImageDetails, err := c.DescribePublic(&ecrpublic.DescribeImagesInput{
+		RepositoryName: aws.String(repository),
+		ImageIds:       imagelookup,
+	})
+	if err != nil {
+		if strings.Contains(err.Error(), "does not exist within the repository") == true {
```

Comment:
In general, matching errors based on strings like this is a no-no. After the release, let's fix this up.